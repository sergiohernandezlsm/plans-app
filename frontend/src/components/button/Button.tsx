import React from "react";
import { SubscriptionSelectedTypes } from '../../types';
import Button from 'react-bootstrap/Button';
import styles from './Button.module.css';

export interface Props {
  planCost?: number;
  selected: boolean;
  contact: string;
  selectionHandler: (x: SubscriptionSelectedTypes) => void;
  id?: number;
}

const ButtonPlan: React.FC<Props> = ({ planCost, selected, contact, selectionHandler, id }) => {
  return (
    <Button onClick={() => selectionHandler({ planValue: planCost, id: id, planType: contact })} className={styles['button-styles']} variant={`${selected ? 'primary' : 'secondary'}`}>{`${contact}: Â£${planCost}`}</Button>
  )
};

export default ButtonPlan;
