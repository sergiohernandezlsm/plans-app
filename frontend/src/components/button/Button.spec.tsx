import { shallow } from 'enzyme';
import Button from './Button';

const props = {
  planCost: 1,
  selected: true,
  contact: 'string',
  selectionHandler: () => { },
  id: 1,
}

describe('Button Component', () => {
  it('renders <Button /> component', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});