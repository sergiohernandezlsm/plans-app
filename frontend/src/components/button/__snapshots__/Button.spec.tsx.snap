// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders <Button /> component 1`] = `ShallowWrapper {}`;
